cmd/arc_summary/arc_summary3

```diff
@@ -221,6 +223,32 @@ elif sys.platform.startswith('linux'):
 
         return descs
 
+def handle_unraisableException(exc_type, exc_value=None, exc_traceback=None,
+                               err_msg=None, object=None):
+   handle_Exception(exc_type[0], exc_type[1], exc_traceback)
+
+def handle_Exception(ex_cls, ex, tb):
+    if ex is IOError:
+        if ex.errno == errno.EPIPE:
+            sys.exit()
+    if ex is KeyboardInterrupt:
+        sys.exit()
+
+    if ex_cls is BrokenPipeError:
+      devnull = os.open(os.devnull, os.O_WRONLY)
+      os.dup2(devnull,sys.stdout.fileno())
+      os.dup2(devnull,sys.stderr.fileno())
+      # It turns out that while sys.exit() triggers an exception
+      # not handled message on Python 3.8+, os._exit() does not.
```

Comment:
You don't need this block (tested on 3.7.3 and 3.9.2) at all (I was also unable to reproduce the error message with sys.exit(), but who knows); the comment may be better served as "Use os._exit() to prevent flushing buffers, which could reraise this" or something maybe

Reply:
So I don't - it's based on [the suggested way of muting SIGPIPE](https://docs.python.org/3/library/signal.html#note-on-sigpipe), and did indeed work to mute the printed backtrace, though not the error code, from the error I was seeing on Python 3.9.2.

[Here](https://imgur.com/a/oFrAHXo) is a screenshot of what I see if I replace os._exit(0) with sys.exit(0) after commenting out those 3 lines and piping it into head -n 10.

Reply:
Hm, reading the error you get I do tend to agree WRT the error causation in your comment, but, oddly, the reason that's noted in the documentation *is* sys.exit -> flush -> reraise. Normal day in Python land, I guess?

Reply:
Who knows? The documentation doesn't necessarily seem inconsistent, since it does say "If an error occurs in the cleanup after the Python interpreter has caught SystemExit (such as an error flushing buffered data in the standard streams), the exit status is changed to 120.", which is indeed what we're seeing here - it's just not that example case.